tests/test_build_latex.py

```diff
@@ -1447,36 +1447,44 @@ def test_latex_labels(app, status, warning):
             r'\label{\detokenize{index:id1}}'
             r'\label{\detokenize{index:figure2}}'
             r'\label{\detokenize{index:figure1}}'
+            r'\sphinxnameddest{\detokenize{figure1}}'
```

Comment:
rst source has two explicit target names 'figure1' and 'figure2'. The code in `hypertarget_to()` will insert only 1 in LaTeX output. Here it is `figure1` which is first in source but I don't know if always the case. I am a currently a bit confused about hyperref explanations about target names, and right now I see why that the latex code is compatible only to define  one name for a given destination and I inherited that from hyperref similar code for "destlabel" option. Although I understand two same names can not point to two distinct targets, I am a bit confused about why a common target could not have two distinct names.

Reply:
This comment is obsoleted from update code